src/basics.md

```diff
@@ -1242,6 +1243,53 @@ fn main() {
 }
 ```
 
+[ex-datetime-arithmetic]: #ex-datetime-arithmetic
+<a name="ex-datetime-arithmetic"></a>
+## Add or subtract dates and times using checked arithmetic
+[![chrono-badge]][chrono] [![cat-date-and-time-badge]][cat-date-and-time]
+
+Calculates and displays the date and time two weeks from now using
+[`DateTime::checked_add_signed`] and the date of yesterday using
+[`DateTime::checked_sub_signed`]. The methods return None if the date and time
+cannot be calculated.
+
+Escape sequences that are available for the
+[`DateTime::format`] can be found at [`chrono::format::strftime`].
+
+```rust
+extern crate chrono;
+use chrono::{DateTime, Duration, Utc};
+
+fn yesterday(date_time: DateTime<Utc>) -> Option<DateTime<Utc>> {
+    date_time.checked_sub_signed(Duration::days(1))
+}
+
+fn main() {
+    let now = Utc::now();
+    println!("{}", now);
+
+    let two_weeks_from_now = now.checked_add_signed(Duration::weeks(2));
+    match two_weeks_from_now {
+        Some(x) => println!("{}", x),
+        None => println!("Two weeks from now overflows!"),
+    }
+
+    let almost_three_weeks_from_now =
+            two_weeks_from_now.unwrap()
```

Comment:
Lets not use `unwrap` in our examples this is exactly why `and_then` was introduced :+1: 

```rust
let almost_three_weeks_from_now = now.checked_add_signed(Duration::weeks(2))
        .and_then(|in_2weeks| in_2weeks.checked_add_signed(Duration::weeks(1)))
        .and_then(day_earlier);
```